nearswap/src/storage_management.rs

```diff
@@ -0,0 +1,161 @@
+        Promise::new(account_id.clone()).transfer(amount);
+        self.storage_balance_of(account_id.try_into().unwrap())
+            .unwrap()
+    }
+
+    #[allow(unused_variables)]
+    fn storage_unregister(&mut self, force: Option<bool>) -> bool {
+        assert_one_yocto();
+        let account_id = env::predecessor_account_id();
+        if let Some(account_deposit) = self.deposits.get(&account_id) {
+            assert!(
+                account_deposit.tokens.is_empty(),
+                "ERR_STORAGE_UNREGISTER_TOKENS_NOT_EMPTY"
+            );
+            self.deposits.remove(&account_id);
+            Promise::new(account_id.clone()).transfer(account_deposit.near);
+            true
+        } else {
+            false
+        }
+    }
+
+    fn storage_balance_bounds(&self) -> StorageBalanceBounds {
+        StorageBalanceBounds {
+            min: U128(MIN_ACCOUNT_DEPOSIT_LENGTH * (env::storage_byte_cost())),
+            max: None,
+        }
+    }
+
+    // check if a user is registered by calling
+    fn storage_balance_of(&self, account_id: ValidAccountId) -> Option<StorageBalance> {
+        if self.deposits.contains_key(account_id.as_ref()) {
+            let acc_deposits = self
+            .deposits
+            .get(account_id.as_ref())
+            .unwrap_or_default();
+            return Some(StorageBalance {
+                total: U128(acc_deposits.near),
+                available: U128(acc_deposits.near - acc_deposits.storage_usage()),
+            })
+        } else {
+            return None;
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::StorageManagement;
+    use super::*;
+
+    use near_sdk::test_utils::{accounts, VMContextBuilder};
+    use near_sdk::{testing_env, BlockHeight, MockedBlockchain};
+
+    fn init_blockchain() {
+        let context = VMContextBuilder::new();
+        testing_env!(context.build());
+    }
+
+    fn new_near_swap() -> NearSwap {
+        let ac = AccountDeposit {
+            near: 9900000000000000000000,
+            storage_used: 10,
+            tokens: HashMap::new(),
+        };
+
+        let mut near = NearSwap {
+            fee_dst: "owner".to_string(),
+            owner: "owner".to_string(),
+            
+            pools: UnorderedMap::new("p".into()),
+
+            deposits: LookupMap::new("d".into()),
+        };
+
+        near.deposits.insert(&"owner".to_string(), &ac);
+
+        return near;
+    }
+
+    #[test]
+    fn storage_balance_works() {
+        init_blockchain();
+
+        let near_swap = new_near_swap();
+
+        StorageManagement::storage_balance_of(&near_swap, "owner".to_string().try_into().unwrap()).unwrap();
+
+        // unwrapping this value will result in error because `owner1` is not in deposits map
+        // therefore function returns None
+        StorageManagement::storage_balance_of(&near_swap, "owner1".to_string().try_into().unwrap());
```

Comment:
please add a check for the returned values.

Reply:
added